torch/distributed/_tensor/ops/embedding_ops.py

```diff
@@ -69,17 +77,23 @@ class _MaskPartial(Partial):
     lifecycle, i.e. the indices_mask would only be alive during the lifetime of the DTensor.
     """
 
-    logical_dim_size: int = -1
     mask_buffer: MaskBuffer = field(default_factory=MaskBuffer)
 
+    # required fields for computing the local offset and deriving the mask
+    offset_shape: Optional[torch.Size] = None
```

Comment:
If this is a required field, it should be non-optional right?

Reply:
@wanchaol 
The filed has to have a default value because it inherits `Partial` which already has a field with default value.
If not None, what default should we use?

Reply:
Oh I see, this make sense, nvm then